sycl/plugins/level_zero/pi_level_zero.hpp

```diff
@@ -723,11 +727,26 @@ struct _pi_queue : _pi_object {
     // Level Zero command queue handles.
     std::vector<ze_command_queue_handle_t> ZeQueues;
 
+    // Immediate commandlist handles, one per Level Zero command queue handle.
+    // These are created only once, along with the L0 queues (see above)
+    // and reused thereafter.
+    std::vector<pi_command_list_ptr_t> ImmCmdLists;
+
+    // Return the index of the next queue to use based on a
+    // round robin strategy and the queue group ordinal.
+    uint32_t getQueueIndex(uint32_t *QueueGroupOrdinal, uint32_t *QueueIndex);
+
+    // Return a queue descriptor using the provided ordinal and index.
+    void createQueueDesc(uint32_t Index, uint32_t Ordinal,
+                         ZeStruct<ze_command_queue_desc_t> &ZeCommandQueueDesc);
```

Comment:
Let's combine the 2 to be `getQueueDescriptor` and return queue descriptor structure.

Reply:
Note that the queue descriptor is not needed each time an index is retrieved, but only the first time, when a commandlist at that index is created.
That's why I think it should be left as it is. If creation of the queue descriptor is in a combined function then either a) the queue descriptor will be created each time a queue index is retrieved, which will be many times because the queues are created only once, or b) the getQueueIndex function will need to know if the queue descriptor is going to be needed or not. The check for "needed or not" is different based on whether imm or std commandlists are being created. So the function will have a weird if-stmt that looks somewhat like:
if ((UseImmCmdlist Imm && CmdLists[Index] != Queue->CommandListMap.end())  || 
     (!UseImmCmdLists && ZeQueues[Index] == nullptr)) ... then create queue descriptor.

Reply:
The queue descriptor is just a small (input is small like few integers) struct, so let's just always create it, even return it even if not needed. Or, if you prefer let's created descriptor in every place it is needed (2 of them), but remove the `createQueueDesc` because it is not encapsulating a reasonable amount of work/semantics.

Reply:
OK